### PR TITLE
Fix #84416 : Allow long note groups to scroll or pan in Create Time Signature dialog

### PR DESCRIPTION
**Pull request by AntonioBL**

### CODE REVIEW

**Comment by IsaacWeiss** [comment]
@lasconic: @duck57 and JGitar requested this again at https://musescore.org/en/user/101731/blog/2016/05/01/musescore-3.0-under-development-musescore-gets-smart#comment-527681, and the PR is still conflict-free and ready to merge eight months later.

**Comment by lasconic** [comment]
@IsaacWeiss Meaning you tried it?

File: mscore/exampleview.cpp [hunk 1]
```diff
@@ -341,6 +365,68 @@ QSize ExampleView::sizeHint() const
...
+            QPointF offsetPt(_matrix.dx(), 0);
+            QRectF firstPageRect(firstPage->pos().x() * _matrix.m11(),
+                                      firstPage->pos().y() * _matrix.m11(),
+                                      firstPage->width() * _matrix.m11(),
+                                      firstPage->height() * _matrix.m11());
+            QRectF lastPageRect(lastPage->pos().x() * _matrix.m11(),
+                                         lastPage->pos().y() * _matrix.m11(),
+                                         lastPage->width() * _matrix.m11(),
+                                         lastPage->height() * _matrix.m11());
+            QRectF pagesRect = firstPageRect.unite(lastPageRect).translated(offsetPt);
```
Comment (lasconic) [inline mscore/exampleview.cpp]: unite is deprecated in Qt >4.8. We can use united.

**Comment by IsaacWeiss** [comment]
Actually, no. I would have guessed somebody else would have, and 
reported a problem before now if there was one. Compilation takes a long 
time on my computer, but I'll give it a go.

**Comment by lasconic** [comment]
Replacing unite by united, I compiled this PR. And I can't see the measure entirely.

capture d ecran 2016-07-08 16 36 17

**Comment by AntonioBL** [comment]
In the PR I was simply copying the part of code from scoreview.cpp responsible for the panning of the score. Maybe with the recent changes it must be tweaked a little (I think the vertical shift is due to these changes; a comparison with scoreview.cpp should highlight the problem).
If I remember correctly, one of the defects I found in the implementation I did here is that if you pan completely to one side and then come back, it does not go to the initial position, but it is shifted by a few pixels.
If I have time in the weekend I can try to update the PR.

**Comment by lasconic** [comment]
@ABL great. Thanks!

**Comment by AntonioBL** [comment]
Sorry that I am late in answering, but I was sick this weekend.
First, a note: the exampleview shown by Nicolas looks like in that picture even without this PR. Indeed, this PR only adds the horizontal scrolling functionality to the exampleview. However, the boundary limits are no more correct with the new changes. I am trying to identify the new values needed in this case; hopefully, I will also find where the vertical offset is set and correct it.
It will take some time (unfortunately I have little free time available).

**Comment by AntonioBL** [comment]
I tried to tackle the problem that were present.
I don't know if it works correctly for high DPI display (and GUI scaling). How can I check this?

**Comment by AntonioBL** [comment]
Rebased to fix merge conflicts

**Comment by AntonioBL** [comment]
And rebased again to fix merge conflicts.

**Comment by ericfont** [comment]
I just tested this, rebased to latest master, and I find that the horizontal scrolling works great, except the calculation of the leftmost & rightmost limit needs to be properly calculated based on the width of the ExampleScore's current width, especially when resizing the window.  Other than that, I support this PR.

I also would like if could scroll via Shift->MiddleMouseScrollWheel, just like I can do with regular score, if my mouse cursor is within the example score rectangle.

**Comment by ericfont** [comment]
basically my rules would be: 
1. leftmost edge of start of staff shouldn't be able to be further right than the margin of the left boundary.
2. rightmost edge of end of staff shouldn't be able to be further left than the margin of the right boundary, unless above condition needs to be met.

**Comment by ericfont** [comment]
As I said, I found AntonioBL's PR allowed way too much scrolling and didn't behave nicely when the widget is resized.  So I added [a commit](https://github.com/musescore/MuseScore/pull/3007/commits/045a255cec648d3ffe8ee8475b14f2d34c5c8e47) to AntonioBL's PR to obey my stricter rules and submitted this as a seperate PR #3007.

**Comment by lasconic** [comment]
See PR #3007.

**Pull request closed without merging**